Adyen.Test/HeaderRequestTest.cs

```diff
@@ -30,90 +33,81 @@ namespace Adyen.Test
     [TestClass]
     public class HeaderRequestTest
     {
-
-        private readonly HttpUrlConnectionClient _httpUrlConnectionClient;
         private readonly string _endpoint = "https://endpoint:8080/";
-
-        public HeaderRequestTest()
-        {
-            _httpUrlConnectionClient = new HttpUrlConnectionClient();
-        }
+        
         [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException), "The given header was not found.")]
         public void BasicAuthenticationHeadersTest()
         {
-            var httpWebRequest = _httpUrlConnectionClient.GetHttpWebRequest(_endpoint, MockPaymentData.CreateConfingMock(), false);
-
-            Assert.IsNotNull(httpWebRequest.UserAgent);
-            Assert.AreEqual(httpWebRequest.Address.ToString(), _endpoint);
-            Assert.AreEqual(httpWebRequest.Headers["Accept-Charset"], "UTF-8");
-            Assert.AreEqual(httpWebRequest.Headers["Cache-Control"], "no-cache");
-            Assert.AreEqual(httpWebRequest.ContentType, "application/json");
-
-            Assert.IsNotNull(httpWebRequest.Headers["Authorization"]);
-            Assert.IsTrue(httpWebRequest.UseDefaultCredentials);
-
-            Assert.AreEqual(httpWebRequest.Headers["Cache-Control"], "no-cache");
-            Assert.IsNull(httpWebRequest.Headers["x-api-key"]);
+            var client = new HttpClientWrapper(MockPaymentData.CreateConfingMock(), new System.Net.Http.HttpClient());
+            var httpRequestMessage = client.GetHttpRequestMessage(_endpoint, false, "requestBody", null, HttpMethod.Post);
+            
+            Assert.IsNotNull(httpRequestMessage.Headers.UserAgent);
+            Assert.AreEqual(httpRequestMessage.RequestUri?.ToString(), _endpoint);
+            Assert.AreEqual(httpRequestMessage.Headers.AcceptCharset.ToString(), "UTF-8");
+            Assert.AreEqual(httpRequestMessage.Headers.CacheControl?.ToString(), "no-cache");
+            Assert.AreEqual(httpRequestMessage.Content.Headers.GetValues("Content-Type").FirstOrDefault(), "application/json; charset=utf-8");
+            Assert.IsNotNull(httpRequestMessage.Headers.Authorization);
+            httpRequestMessage.Headers.GetValues("x-api-key");
```

Comment:
Recommendation, use: Assert.IsFalse(httpRequestMessage.Headers.TryGetValues("x-api-key", out var _));

Reply:
Remove: [ExpectedException(typeof(InvalidOperationException), "The given header was not found.")]